app/src/main/java/org/kiwix/kiwixmobile/nav/destination/reader/KiwixReaderFragment.kt

```diff
@@ -250,9 +251,13 @@ class KiwixReaderFragment : CoreReaderFragment() {
     val settings = requireActivity().getSharedPreferences(SharedPreferenceUtil.PREF_KIWIX_MOBILE, 0)
     val zimFile = settings.getString(TAG_CURRENT_FILE, null)
 
-    if (zimFile != null) {
+    if (zimFile != null && File(zimFile).exists()) {
       if (zimReaderContainer.zimFile == null) {
         openZimFile(File(zimFile))
+        Log.d(
+          TAG_KIWIX,
+          "Kiwix normal start, zimFile loaded last time -> Open last used zimFile $zimFile"
```

Comment:
You can just say
```suggestion
          "Opened last used zimFile: $zimFile"
```

Reply:
_Kiwix normal start_
We need this piece to differentiate between normal launches and launches via deepLinks or fileManagers, yeah the second part can be changed as per suggestion.

Reply:
3748b17